src/components/TsmlUI.tsx

```diff
@@ -277,6 +296,7 @@ export default function TsmlUI({
               inProgress={inProgress}
               setState={setState}
               state={state}
+              setSearchParams={setSearhParams}
```

Comment:
i would take out the searchParams, setSearchParams, and setState props from all sub components - `Alert`, `Title`, `Controls`, `Table`, `Map` - setState becomes something that only the `TsmlUi` component can do